### ISSUE TITLE
Flippers intermittent lock up or don’t respond

### ISSUE THREAD

**Issue by mazjohn** [vpinball/pinmame#262]
Flippers fail to fire or intermittently lock up… the issue Occurs with VpinMame build 917 with the following table written for 10.7,  run with Vpx X64 build 935 running in 10.8 window mode.

andromeda v4.0
https://www.vpforums.org/index.php?app=downloads&showfile=14871

I am only reporting this since I recall  there was a recent flipper issue recently resolved/ closed with WOF. The issue didn’t happen for a game or two, then started to happen quite a bit where the flippers randomly stay stuck in the engaged position or sometimes intermittently fail to trigger. This Also happened again eventually after restarting the table.  This  is a 10.74 table so if it’s not intended to be compatible on 10.8, no worries at all. …thank you.

**Comment by francisdb** [comment]
Link to possible related mentioned issue #1536 

I could not reproduce this here (on standalone). Even without adding the `vpmInit Me`.

**Comment by toxieainc** [comment]
@mazjohn Please retest, this should be fixed by now

**Comment by mazjohn** [comment]
Left flipper stuck engaged until pressing left flipper button then eventually released with following table… Vpx x64 build 1942 and vpinmame build 919…. https://vpuniverse.com/files/file/4626-lord-of-the-rings-hanibals-4k-edition…. happened a few times after coin entered before ball launch… game was very playable otherwise..

**Comment by vbousquet** [comment]
Perhaps updating B2S to the latest version (which contains update regarding latest PinMame change) may help : [B2S 2.1.1](https://github.com/vpinball/b2s-backglass/releases/tag/v2.1.1)

**Comment by WaRcLaWz** [comment]
> Perhaps updating B2S to the latest version (which contains update regarding latest PinMame change) may help : [B2S 2.1.1](https://github.com/vpinball/b2s-backglass/releases/tag/v2.1.1)

I will try that out and report back my findings. I have a few tables that will do it immediately as soon as I switch back to EXE mode :)

**Comment by toxieainc** [comment]
@JockeJarre for vis

**Comment by sverrewl** [comment]
I see that you mention different architectures being tested, so, I don't expect that this will be related. But, from years of experience. Starting over with a fresh nvram has often solved this problem on my end.

**Comment by JockeJarre** [comment]
Sorry, I cannot see this behaviour in any of the releases

**Comment by mazjohn** [comment]
Fast forward, quick update since submitting this bug-  as oftodays date running latest B2s server release 212 with current Vpx 10.8  x32/64 beta4, combined with a currentvpinmame sc1001 build, have not seen the flipper lockup issue return in random play. I will be reloading some of the past tables that had the stuck flipper issue to further verify…. I can also confirm, The andromeda table was fixed with the above patch

**Comment by michroch** [comment]
I got the same problem on some tables, the problem of slow flipper and flipper stuck up is when I double click the table directly in the tables file. If starting VPX program first then choose the table in question, everything is fine.

**Comment by Ltek** [comment]
I'm working on a table based on the Swords of Fury ROM, and it has major issues with sticking flippers. It has 

I've seen a few posts saying talking about PinMAMETimer.Interval not making =1 but that is not helpful at all since it doesnt state the proper interval?

What are the correct settings?

Currently...

```
	PinMAMETimer.Interval = -1
	PinMAMETimer.Enabled = 0
	On Error Resume Next
	vpmInit Me

```

**Comment by sverrewl** [comment]
I think that this url has a lot of examples for how to script tables.

https://github.com/sverrewl/vpxtable_scripts

More good info that everyone fiddling with table scripts should read

https://www.vpforums.org/index.php?showtopic=4261

Obviously, in the few lines you've shown, you are requesting and interval that is never enabled. Doesn't really makes sense, just from those lines - does it ?

Not a pinmame issue.

**Comment by Ltek** [comment]
@sverrewl this issue is posted as a bug, by others stating it has happened with many ROM tables. It may not be a VPX bug, it may be VPM, who knows but I added additional evidence (including a full table) that this bug always shows itself within a min or two of play, and just keeps happening. I and VPW testers have seen this table show sticky flippers constantly... not just a little, a lot.

also re: PinMAMETimer.Interval = -1 ... I dont know if the setting is causing it. -1 is the recommended setting from VPW team. I was only stating it is the setting I'm using (as does all VPW tables) and asking if that is somehow causing the issue -and- what is the recommended setting from the VPX devs... this is the correct place to ask VPX expert developer-level questions.

**Comment by sverrewl** [comment]
Yeah. Well, @vbousquet didn't tell you to disable the flipper keys when it hits the drain. How are you supposed to play mb on this mod ? Like I said. Issue belongs to regular forums.

**Comment by Ltek** [comment]
> Yeah. Well, @vbousquet didn't tell you to disable the flipper keys when it hits the drain. How are you supposed to play mb on this mod ? Like I said. Issue belongs to regular forums.

irrelevant to the  bug this thread is about and not what I've asked about. I dont understand why you are complaining that I posted an example of a bug other people have validated exists.

**Comment by sverrewl** [comment]
mb - ok, short for multiball, which this machines has. You posted a this as a bug, instead of understanding your own mistake. It has nothing to do with the topic, except that you purposely are disabling the flippers - forcefully making the flippers lock up yourself.

VPX and Pinmame is complex software. Requires a lot of time from volunteers, we should all strive to let them code instead of having to shift focus onto things that isn't valid.

I haven't checked the other reported problems in this thread, they might be valid.

**Comment by Ltek** [comment]
mazjohn posted this bug back in March, and others posted it in other tickets. I simply added to it.

If you know something I dont please point to the exact code causing my problem, otherwise you are just trolling.

If you mean the code where I disabled flipper keycodes, that doesnt matter I have since disabled ALL the flipper keycode and it works, yet still has the stuck flipper bug.

This is a ROM table so flippers are triggered with Solenoids callbacks. The flipper work without keycodes via cab buttons -and- keyboard. I know because I play it on the cab -and- test via remote desktop with keyboard.

If you can real help do so, it would be appreciated... or stop posting if you cannot.

**Comment by sverrewl** [comment]
Best take this back into forums. It is your code that is faulty. I told you, you are setting a flag - preventing key up and down to work. This gives non responsive flippers in either up or down position. I recommend that you look the latest release of Sword of Fury for properly set timers and through usage.

I haven't spent more than a minute figuring out what the issue is on your mod, and I haven't looked in details to the latest release for that table either. All I can say, it is best to study work from Sixtoe, Apophis or any of those VPW magicians on how to proper create a table - deleting and adding balls is old school stuff and has and still is often a root cause for oddities.

Again, your table mod is not related to the rom, or should I say, it could be - but, you are masquerading it by your code.

**Comment by vbousquet** [comment]
Regarding this issue, I think there have been a few different things that may make things difficult to understand. On my side, my understanding is the following:
- there used to be a combination of VPX/VPM/B2S version that would cause this issue. So first, these needs to be updated to latest (at the time of typing this message: RC4 for VPX, 2.1.2 for B2S, latest master for VPM),
- the way input is handled is somewhat hacky for VPX **GL** version (2 message pump running in parallel inside VPX eventually stealing messages from each other, with a message pump running as well inside VPM for fast flips). This needed some deeper change, so I do not plan to fix it for VPX 10.8 (was already like that in VPVR) and the fix for this would be in 10.8.1 (already in, but 10.8.1 is in pre-alpha state and I would strongly advise against using it).

If I'm not missing anything, I would say that the situation (with a known bug for VPX GL) is ok for the 10.8 release. If it's something else, it would need to be looked at.

**Comment by Ltek** [comment]
I'm getting stuck flippers with DX build. Interestingly, happen less often with nFozzy flipper code than it did with JPSs.
My system...
- VPX 10.8 build 1975 DX64
- VPM 3.6.0.988
- No DirectB2S file running at all, uses PuP Pack.
- Launching directly with VPX, not popper
- happens with both Full Screen Exclusive and Window
- happened with both JPS's original flipper code, and with nFozzy (added ~12 days ago)
- happens with staged flippers and non-staged (staged flipper code just added 2-3 days ago)

If you want to see it first hand (latest build, beta 23)...
https://mega.nz/file/Vz0AiLZb#ZmADiqoEA9j6SbBfjqKBpWE45Vty8AmMXcU0LuQY0iA

**Comment by Loadedweapon** [comment]
yea this is a very annoying and intermittent bug. I have been working on a ton of tables just with the editor no backglass or puppacks and this happens often. I might play the table 1 time perfect then another flippers randomly stick up till I press it again. Sometimes it's as soon as I start playing sometimes never happens. But even the latest 1167 does it and multiple people in my discord having the same issue. I know for a fact that 1108 never does it most people have rolled back to that but it breaks other updates like the fix for MM draw bridge.

**Comment by mazjohn** [comment]
This issue is back more prevalent than ever...over 50-100 tables with flippers getting stuck in 10.80 final w/vpinmame 10.80 final build 1227?
